docs/developer-guide/api/concrete.ml.sklearn.base.md

```diff
@@ -2946,7 +3118,631 @@ predict_proba(
+#### <kbd>property</kbd> fhe_circuit
+
+Get the FHE circuit.
+
+The FHE circuit combines computational graph, mlir, client and server into a single object. More information available in Concrete documentation (https://docs.zama.ai/concrete/getting-started/terminology_and_structure) Is None if the model is not fitted.
+
+**Returns:**
+
+- <b>`Circuit`</b>:  The FHE circuit.
+
+______________________________________________________________________
+
+#### <kbd>property</kbd> is_compiled
+
+Indicate if the model is compiled.
+
+**Returns:**
+
+- <b>`bool`</b>:  If the model is compiled.
+
+______________________________________________________________________
+
+#### <kbd>property</kbd> is_fitted
+
+Indicate if the model is fitted.
+
+**Returns:**
+
+- <b>`bool`</b>:  If the model is fitted.
+
+______________________________________________________________________
+
+#### <kbd>property</kbd> onnx_model
+
+Get the ONNX model.
+
+Is None if the model is not fitted.
+
+**Returns:**
+
+- <b>`onnx.ModelProto`</b>:  The ONNX model.
+
+______________________________________________________________________
+
+<a href="../../../src/concrete/ml/sklearn/base.py#L317"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+### <kbd>method</kbd> `check_model_is_compiled`
+
+```python
+check_model_is_compiled() → None
+```
+
+Check if the model is compiled.
+
+**Raises:**
+
+- <b>`AttributeError`</b>:  If the model is not compiled.
+
+______________________________________________________________________
+
+<a href="../../../src/concrete/ml/sklearn/base.py#L293"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+### <kbd>method</kbd> `check_model_is_fitted`
+
+```python
+check_model_is_fitted() → None
+```
+
+Check if the model is fitted.
+
+**Raises:**
+
+- <b>`AttributeError`</b>:  If the model is not fitted.
+
+______________________________________________________________________
+
+<a href="../../../src/concrete/ml/sklearn/base.py#L2068"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+### <kbd>method</kbd> `compile`
+
+```python
+compile(*args, **kwargs) → Circuit
+```
+
+______________________________________________________________________
+
+<a href="../../../src/concrete/ml/sklearn/base.py#L1882"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+### <kbd>method</kbd> `dequantize_output`
+
+```python
+dequantize_output(q_y_preds: 'ndarray') → ndarray
+```
+
+______________________________________________________________________
+
+<a href="../../../src/concrete/ml/sklearn/base.py#L228"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+### <kbd>method</kbd> `dump`
+
+```python
+dump(file: 'TextIO') → None
+```
+
+Dump itself to a file.
```

Comment:
is the location important?

Reply:
the file is passed as an argument as specified in the args list